backend/app.py

```
@@ -54,7 +54,8 @@ def process_query():
         return jsonify(result)
 
     except Exception as e:
-        return jsonify({"error": str(e)}), 500
+        app.logger.error(f"Error processing query: {e}")
```

Ruff (0.8.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)